src/Http/Routing/src/HttpMethodMetadata.cs

```diff
@@ -41,10 +41,60 @@ public HttpMethodMetadata(IEnumerable<string> httpMethods, bool acceptCorsPrefli
                 throw new ArgumentNullException(nameof(httpMethods));
             }
 
-            HttpMethods = httpMethods.ToArray();
+            HttpMethods = httpMethods.Select(GetHttpMethod).ToArray();
             AcceptCorsPreflight = acceptCorsPreflight;
         }
 
+        private static string GetHttpMethod(string httpMethod)
```

Comment:
Should this method be in the HttpMethods class? It seems like other components could use it.